components/playground/instance/instance.go

```diff
@@ -70,6 +70,8 @@ type Instance interface {
 	// Wait Should only call this if the instance is started successfully.
 	// The implementation should be safe to call Wait multi times.
 	Wait() error
+	// BinPathCheck return the bin path to check component version.
+	BinPathCheck(utils.Version) string
```

Comment:
How about simply return the bin path (and call it `BinPath`)? The current API mixes two different things together, which is confusing and not useful for other purposes.

Reply:
How about advancing the `PrepareBinary` function?